### PR TITLE
Add an option to add newlines to the end of files.

### PR DESCRIPTION
**Pull request by jkinkead**
Fixes https://github.com/scala-ide/scalariform/issues/113 .

### CODE REVIEW

File: scalariform/src/test/scala/scalariform/formatter/NewlineAtEndOfFileFormatterTest.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  // Must parse as a full script to verify the newline formatting.
+  def parse(parser: ScalaParser) = parser.scriptBody()
+
+  def format(formatter: ScalaFormatter, result: Result) = formatter.format(result)(FormatterState())
+
+  override val debug = false
+
+  {
+  implicit val formattingPreferences = FormattingPreferences.setPreference(NewlineAtEndOfFile, true)
```
Comment (sschaef) [inline scalariform/src/test/scala/scalariform/formatter/NewlineAtEndOfFileFormatterTest.scala]: The tests should be repeated for the false case.

Comment (jkinkead) [inline scalariform/src/test/scala/scalariform/formatter/NewlineAtEndOfFileFormatterTest.scala]: Done!

Comment (jkinkead) [inline scalariform/src/test/scala/scalariform/formatter/NewlineAtEndOfFileFormatterTest.scala]: Note that I had to rename them due to how the test cases are generated - they need different text, else they end up with the same name & cause errors while running.

**Comment by sschaef** [comment]
LGTM

**Pull request merged**